src/gwc/src/main/java/org/geoserver/gwc/layer/DefaultTileLayerCatalog.java

```diff
@@ -227,9 +254,72 @@ public GeoServerTileLayerInfo save(final GeoServerTileLayerInfo newValue) {
         return oldValue;
     }
 
+    private GeoServerTileLayerInfoImpl load(Resource res) {
+        GeoServerTileLayerInfoImpl info;
+        try {
+            info = depersist(res);
+            startToListen(res, info.getId());
+        } catch (Exception e) {
+            LOGGER.log(
+                    Level.SEVERE,
+                    "Error depersisting tile layer information from file " + res.name(),
+                    e);
+            return null;
+        }
+
+        layersByName.put(info.getName(), info.getId());
+        layersById.put(info.getId(), info);
+
+        if (LOGGER.isLoggable(Level.FINER)) {
+            LOGGER.finer("Loaded tile layer '" + info.getName() + "'");
+        }
+
+        return info;
+    }
+
+    private void reload(String id, Resource res) {
+        GeoServerTileLayerInfo old = layersById.remove(id);
+        if (old != null) {
+            layersByName.remove(old.getName());
+        }
+        load(res);
+    }
+
+    private void startToListen(Resource file, String tileLayerId) {
+        ResourceListener existingLayerListener =
+                new ResourceListener() {
+                    @Override
+                    public void changed(ResourceNotification notify) {
+                        if (notify.getKind() == ResourceNotification.Kind.ENTRY_MODIFY) {
+                            reload(tileLayerId, resourceLoader.get(notify.getPath()));
+                            for (TileLayerCatalogListener listener : listeners) {
+                                listener.onEvent(tileLayerId, TileLayerCatalogListener.Type.MODIFY);
+                            }
+                        } else if (notify.getKind() == ResourceNotification.Kind.ENTRY_DELETE) {
+                            delete(tileLayerId);
+                            for (TileLayerCatalogListener listener : listeners) {
+                                listener.onEvent(tileLayerId, TileLayerCatalogListener.Type.DELETE);
+                            }
+                        }
+                    }
+                };
+        listenersByFileName.put(file.name(), existingLayerListener);
+        file.addListener(existingLayerListener);
+    }
+
+    private void stopToListen(Resource file) {
```

Comment:
This is a confusing name.  It suggests some sort of synchronous pause/wait: "stop" in order to "listen". `stopListening` would be clearer.  Renaming the other method `startListening` would keep the pairing between them clear.